contracts/token/ERC677Token.sol

```diff
@@ -49,6 +49,7 @@ contract ERC677Token is IERC677, ERC20PresetMinterPauserUpgradeable {
 
     emit Transfer(msg.sender, _to, _value, _data);
 
+    // Note: isContract() is not guaranteed to return an accurate value, never use it to provide an assurance of security.
```

Comment:
I'm interested in knowing why is it not guaranteed?

Reply:
There are some useful insights here: https://ethereum.stackexchange.com/questions/15641/how-does-a-contract-find-out-if-another-address-is-a-contract